addons/website_event/models/event_question.py

```diff
@@ -23,12 +28,41 @@ class EventQuestion(models.Model):
                                     help="If True, this question will be asked only once and its value will be propagated to every attendees."
                                          "If not it will be asked for every attendee of a reservation.")
     is_mandatory_answer = fields.Boolean('Mandatory Answer')
+    is_identification_question = fields.Boolean('Identification Question', compute='_compute_is_identification_question')
 
     @api.constrains('event_type_id', 'event_id')
     def _constrains_event(self):
         if any(question.event_type_id and question.event_id for question in self):
             raise UserError(_('Question cannot belong to both the event category and itself.'))
 
+    @api.depends('question_type')
+    def _compute_title(self):
+        """ Since title is mandatory and to prevent the user from entering it
+        for identification questions, it will be computed automatically. But we
+        also want it to reset automatically if the question_type changes."""
+        for question in self:
+            question_type_dict = dict(self._fields['question_type']._description_selection(self.env))
+            if question.question_type not in ['simple_choice', 'text_box']:
+                question.title = question_type_dict.get(question.question_type)
+            elif question.title in question_type_dict.values():
+                question.title = False
```

Comment:
Have to say I also don't understand the purpose, this seems rather complicated just for a title.

Would say let people enter what they want and initialize maybe for the "identification questions"  if we want to. See other comment.

(Which makes me wonder if we even need a "is_identification_question" field).